src/github.com/travis-ci/worker/processor_pool.go

```diff
@@ -113,7 +112,31 @@ func (p *ProcessorPool) GracefulShutdown() {
 	}
 }
 
-func (p *ProcessorPool) processor(queue *JobQueue) error {
+// Incr adds a single running processor to the pool
+func (p *ProcessorPool) Incr() {
+	p.processorsWG.Add(1)
+	go func() {
+		err := p.runProcessor(p.queue)
+		if err != nil {
+			p.poolErrors = append(p.poolErrors, err)
+			return
+		}
+		p.processorsWG.Done()
```

Comment:
I think this should be moved into a `defer`, since the `return` two lines above it can cause this line not to be called if there was an error running the processor.